src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -431,13 +431,16 @@ public unsafe void SetResponseHeaders()
                 var knownHeaderIndex = HttpApiTypes.HTTP_RESPONSE_HEADER_ID.IndexOfKnownHeader(headerPair.Key);
                 for (var i = 0; i < headerValues.Count; i++)
                 {
-                    if (string.IsNullOrEmpty(headerValues[i]))
+                    var headerValue = headerValues[i] ?? string.Empty;
```

Comment:
nit: shouldn't need the `?? string.Empty` part, compiler will know `headerValue` isn't null after the `IsNullOrEmpty` check